File: CommonLib/src/main/java/com/enzo/commonlib/widget/tablayout/ITabLayout.java
package com.enzo.commonlib.widget.tablayout;

import java.util.List;

/**
 * 文 件 名: ITabLayout
 * 创 建 人: xiaofangyin
 * 创建日期: 2017/12/17
 * 邮   箱: xiaofangyinwork@163.com
 */
public interface ITabLayout {

    void initData(List<TabEntity> list);

    void setCurrentItem(int currentItem);

    int getCurrentItem();

    void addMessageNum(int position, int messageNum);

    void setMessageNum(int position, int messageNum);

    void resetMessageNum(int position);

    int getMessageNum(int position);

    void showRedPoint(int position);

    void hideRedPoint(int position);

    void showTabLayout();

    void hideTabLayout();
}

File: commonlibkt/src/main/java/com/enzo/commonlibkt/widget/tablayout/ITabLayout.kt
package com.enzo.commonlibkt.widget.tablayout

/**
 * 文 件 名: ITabLayout
 * 创 建 人: xiaofangyin
 * 创建日期: 2017/12/17
 * 邮   箱: xiaofangyinwork@163.com
 */
interface ITabLayout {
    fun initData(list: List<TabEntity>)
    var currentItem: Int
    fun addMessageNum(position: Int, messageNum: Int)
    fun setMessageNum(position: Int, messageNum: Int)
    fun resetMessageNum(position: Int)
    fun getMessageNum(position: Int): Int
    fun showRedPoint(position: Int)
    fun hideRedPoint(position: Int)
}